### PR TITLE
OVDB-41 coord hash v1

### PR DESCRIPTION
**Pull request by kmuseth**

### CODE REVIEW

File: openvdb/math/Coord.h [hunk 1]
```diff
@@ -578,8 +591,24 @@ operator<<(std::ostream& os, const CoordBBox& b)
...
+// template specialization of std::hash with Coord, which
+// allows for Coord to be used as the key in std::unordered_map
+namespace std {// injected in namespace std
+
+template<>
+struct hash<openvdb::math::Coord>
+{
+    using argument_type = openvdb::math::Coord;
+    using result_type = std::size_t;
+    result_type operator()(const argument_type& ijk) const noexcept { return ijk.hash<>(); }
```
Comment (pcucka) [inline openvdb/math/Coord.h]: ICC complains about this (I'm not sure why):
```
openvdb/math/Coord.h(605): warning #955: ambiguous class member reference --
    function template "openvdb::v6_0::math::Coord::hash" (declared at line 257)
    used in preference to type "std::hash<openvdb::v6_0::math::Coord>::hash" (declared at line 601)

    result_type operator()(const argument_type& ijk) const noexcept { return ijk.hash<>(); }
```
I suggest changing it to
```
template<>
struct hash<openvdb::math::Coord>
{
    using Coord = openvdb::math::Coord;
    using argument_type = Coord;
    using result_type = std::size_t;
    result_type operator()(const Coord& ijk) const noexcept { return ijk.Coord::hash<>(); }
};
```

File: openvdb/CHANGES [hunk 2]
```diff
@@ -8,6 +8,8 @@ Version 6.0.1 - In development
       VecTraits and added an IsSpecializationOf helper metafunction.
     - Added support for Vec4s, Vec4d and Vec4i metadata.
     - Added util::printTime that outputs nicely-formatted time information.
+    - Added a template specialization of std::hash with Coord.
```
Comment (pcucka) [inline openvdb/CHANGES]: Please update `doc/changes.txt` as well.

File: openvdb/math/Coord.h (hunk 1 above)
Comment (kmuseth) [reply openvdb/math/Coord.h]: done

File: openvdb/CHANGES (hunk 2 above)
Comment (kmuseth) [reply openvdb/CHANGES]: done

**Pull request merged**